cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -464,12 +464,16 @@ func ValidateAPIEndpoint(c *kubeadm.APIEndpoint, fldPath *field.Path) field.Erro
 }
 
 // ValidateIgnorePreflightErrors validates duplicates in ignore-preflight-errors flag.
-func ValidateIgnorePreflightErrors(ignorePreflightErrors []string) (sets.String, error) {
+// Multiple arrays of ignored pre-flight errors can be provided, as there may
+// be various sources for these, e.g. CLI options, configuration files, etc.
+func ValidateIgnorePreflightErrors(ignorePreflightErrorsArrays ...[]string) (sets.String, error) {
```

Comment:
Ok, this is a little bit different, from what I expected.
The thing is that we want "all" to be an invalid value for errors specified in the config YAML.
Hence, it's beset to keep more or less the previous variant of the func, add a bool parameter to it (to tell if the error list is from the config file or the command line) and then denying "all" if the bool says it's from the config file.

Reply:
@rosti, my suggestion was instead of `...[]string` to have two explicit arguments - one from command line, one from config. then to have two for loops, one processing config, then checking if 'all' is present in set (error if yes), then add overrides from command line. No need for bool here.

Reply:
Sure, I'll go for this then.

Reply:
Note that if we go for two explicit arguments (1), whether at this level, or in a wrapper function, maintainers would have to ensure arguments are passed in the right order.

If we go for a boolean (2), is may be a tiny bit more explicit, but the boolean would still be rather obscure, i.e. one would have to look at its name in the implementation to know its meaning.

A third approach (3), more verbose, but more explicit, would be to have two methods, e.g.`ValidateIgnorePreflightErrorsFromCLI` and `ValidateIgnorePreflightErrorsFromConfigFile` (or so, better names always welcome).

@kad, @rosti, please let me know your preferences.
(At the moment, I'm hesitating between (1) and (3).)

Reply:
(1) currently implemented.